internal/plugins/helm/v1/api.go

```diff
@@ -180,21 +178,42 @@ func (p *createAPISubcommand) Validate() error {
 		if len(strings.TrimSpace(p.createOptions.GVK.Kind)) == 0 {
 			return fmt.Errorf("value of --%s must not have empty value", kindFlag)
 		}
+	}
 
-		// Validate the resource.
-		ogvk := p.createOptions.GVK
-		gvk := resource.GVK{Group: ogvk.Group, Version: ogvk.Version, Kind: ogvk.Kind}
-		if err := gvk.Validate(); err != nil {
-			return err
-		}
+	// Create and validate the resource and chart from CreateOptions.
+	projectDir, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	p.resource, p.chrt, err = chartutil.CreateChart(p.config, projectDir, p.createOptions)
```

Comment:
Or perhaps return something that can write the chart after returning like a function? Creating a temp dir is probably ok but would be leaky in case validation fails.

Reply:
If that's possible, that would also work. But I think you'd want to (and might have to) fetch remote charts to be able to validate them and anything derived from them.

It honestly probably makes sense to do the fetch outside the project directory anyway to validate that the chart is fetchable.

Perhaps we fetch into a temp dir, run the validations and then hand the scaffolder a function that effectively calls `mv $TMPDIR/<chart> $PROJECT_DIR/helm-charts/<chart> && rm -r $TMPDIR`.

Reply:
Instead of returning a function, I just call `chartutil.SaveDir()` in the API scaffolder.

Reply:
I should probably bear this in mind in #4581 as well, as right now it is being created in `InjectResource` and it should probably need to be created in either `PreScaffold` or `Scaffold` hooks (plugin phase 1.5 defines a more extensive set of plugin subcommand hooks, you can check the [design document](https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/extensible-cli-and-scaffolding-plugins-phase-1-5.md) for further info).